### PR TITLE
Make Excel row_handler not default to empty string value

### PR DESCRIPTION
**Pull request by Snotpus**
In python, turns out `0 or ''` results in the empty string...

This was causing an issue when an Excel file was reading in a zero (0) integer value, and the row_handler was squashing away the value

### CODE REVIEW

File: rumydata/table.py [hunk 1]
```diff
@@ -432,7 +432,7 @@ def __exit__(self, exc_type, exc_val, exc_tb):
 
     @staticmethod
     def _row_handler(row: list) -> List[str]:
-        return [str(x or '') for x in row]
+        return [str(x) or '' for x in row]
```
Comment (Mikuana) [inline rumydata/table.py:435]:
```suggestion
        return ['' if x is None else str(x) for x in row]
```

Comment (Mikuana) [reply rumydata/table.py:435]: Doing it this way explicitly checks for the `None`, which is what this handler was primarily looking for in the first place. If we just coerce a None value via `str(None)`, that will cause the row handler to pass empty cells with a value of None instead of keeping them empty.

Comment (Mikuana) [reply rumydata/table.py:435]: I'm realizing now that this `_row_handler` method is specific to the `ExcelFile` class, and getting at what you were saying about needing to run tests through each type. That seems like it could be never ending, but I think we can be fairly target about it. The **biggest** difference that can be introduced by different types is the way that this row handler operates; everything else gets handled by the same code under the row level. So in terms of writing tests, what I think we need to do is write some specific row tests which do this:

1. Create a table which tries to test for the row-level edge cases
2. Write that table into an Excel file (and for that matter, Parquet as well)
3. Test that table for the expected errors

**Comment by Snotpus** [comment]
I expanded with tests where I thought it made sense. Please check it out!

File: tests/utils.py [hunk 2]
```diff
@@ -11,3 +20,43 @@ def func(*args):
...
+                assert not obj.check(p)
+            except AssertionError as e:
+                aes[nm] = e
+        new_line = '\n'
+        assert not aes, f'Write test failed for:\n {new_line.join([f"{k}:{v}" for k, v in aes.items()])}'
+
+
+def file_cell_harness(value: Union[str, list], field: Union[Field, dict]):
+    """ Wrapper to convert cell to row for harness check """
+    if isinstance(value, list):
```
Comment (Mikuana) [inline tests/utils.py:59]: I think this change makes file_cell wrapper equivalent to file_row_harness. Let's try replacing any time this is called for a row with `file_row_harness` instead. If it doesn't blow up, we should revert `file_cell_harness` back to only taking a single value.

File: tests/test_fields.py [hunk 3]
```diff
@@ -224,27 +257,42 @@ def test_static_rules_bad(value, func, assertion):
...
+    value = ('1', {'x': '0'})
     x = field.Field(rules=[rules.cell.GreaterThanColumn('x')])
-    assert not x.check_cell(('1', {'x': '0'}))
+    assert not x.check_cell(value)
+
+
+def test_column_compare_rule_good_files():
+    value = ['1', '0']
+    lay = {'c1': field.Field(rules=[rules.cell.GreaterThanColumn('c2')]), 'c2': field.Integer(1)}
+    file_cell_harness(value, lay)
```
Comment (Mikuana) [inline tests/test_fields.py:274]:
```suggestion
    file_row_harness(value, lay)
```

We should try this to test if we can revert the cell function back to taking only single values.

File: tests/test_fields.py [hunk 4]
```diff
@@ -56,7 +59,10 @@ def test_text_good(value, kwargs):
     ('x', dict(max_length=80, min_length=2), rules.cell.MinChar),
 ])
 def test_text_bad(value, kwargs, rule):
-    assert field.Text(**kwargs)._has_error(value, rule.rule_exception())
+    fld = field.Text(**kwargs)
+    assert fld._has_error(value, rule.rule_exception())
+    with pytest.raises(AssertionError):
```
Comment (Mikuana) [inline tests/test_fields.py:64]: There is a bit of a gap here, where the harness could throw an AssertionError, but only for one of the file types. In the case where we're expecting all files to have an error, this could leave a gap in the testing. This is not a deal breaker, but something to think on for how we might improve the harness in the future.

File: tests/test_fields.py [hunk 5]
```diff
@@ -224,27 +257,42 @@ def test_static_rules_bad(value, func, assertion):
...
+                                            z=field.Ignore()))
 
 
 def test_column_unique_bad():
     x = field.Field(rules=[rules.column.Unique()])
     assert x._has_error(['2', '2'], rules.column.Unique.rule_exception(), rule_type=field.cr.Rule)
+    file_cell_harness(['2', '2'], dict(x=x, y=x))
 
 
 def test_empty_field():
```
Comment (Snotpus) [inline tests/test_fields.py:298]: Add file harness assertion for this test, it's very close to our use case.

**Comment by Snotpus** [comment]
Increment version number

File: tests/test_fields.py (hunk 4 above)
Comment (Snotpus) [reply tests/test_fields.py:64]: I've completed the other pieces of feedback, but I kinda want to take a crack at this. But perhaps we should release what we have so that our users can take advantage of the Excel fix, and I can attempt to rework some of this file harness in a separate 'feature' branch. Does that sound like a plan?

File: tests/utils.py (hunk 2 above)
Comment (Snotpus) [reply tests/utils.py]: Yuuup, you're absolutely right. Reverted `file_cell_harness` and used `file_row_harness` instead and it just works. Smh

File: tests/test_fields.py (hunk 3 above)
Comment (Snotpus) [reply tests/test_fields.py]: Swapped to `file_row_harness` in this and similar tests, works great!

**Comment by Snotpus** [comment]
Hmm I'm not exactly sure why codecov is thinking `test_fields.py` has lost coverage... running it locally shows it as 100%. But maybe pycharm's coverage mechanism differs from how codecov calculates it? :/

**Comment by Snotpus** [comment]
> Hmm I'm not exactly sure why codecov is thinking `test_fields.py` has lost coverage... running it locally shows it as 100%. But maybe pycharm's coverage mechanism differs from how codecov calculates it? :/

Okay, after authing github to codecov, I was actually able to see the details from the codecov report I was expecting to see! I think I've got this fixed up...
